src/openforms/authentication/contrib/digid_eherkenning_oidc/views.py

```diff
@@ -93,6 +102,17 @@ def failure_url(self):
 
         return f.url
 
+    @property
+    def success_url(self):
+        # Pull the next url from the session or settings--we don't need to
+        # sanitize here because it should already have been sanitized.
+        next_url = self.request.session.get("oidc_login_next", None)
+        return (
+            next_url
+            or self._redirect_next
```

Comment:
This should be needed only in the case of a failed login right? So maybe it makes more sense to keep it in the `failure_url` property?

Also, this issue in the Mozilla package seems relevant: https://github.com/mozilla/mozilla-django-oidc/issues/458, maybe we could mention it in a comment?

Reply:
I think something was lost in communication here - IMO the `success_url` should not be touched, instead the `failure_url` property should be updated to make use of `self._redirect_next` instead of relying on `self.success_url`.

It feels wrong and dangerous to mix success/failure behaviours.